.grit/patterns/jest_to_vitest.md

```diff
@@ -0,0 +1,294 @@
+---
+title: Jest to Vitest
+---
+
+Convert Jest tests to Vitest
+
+tags: #migration, #js
+
+```grit
+engine marzano(0.1)
+language js
+
+pattern adjust_imports() {
+  $body where {
+    $body <: or {
+      `$sym.$_($_)`,
+      `$sym($_)`,
+    } where {
+      $program <: program($statements),
+      $mocha = `'mocha'`,
+      $sym <: or {
+        `test`,
+        `expect`,
+        `describe`,
+        `it`,
+        `beforeEach`,
+        `afterEach`,
+        `beforeAll`,
+        `afterAll`,
+        `jest`,
+        `vi`
+      } where {
+        // Filter out methods which are imported from mocha
+        $sym <: not imported_from(from = $mocha),
+        // Filter out methods which are imported as default from any module
+        $statements <: not contains or {
+          `import $sym from $_`,
+        }
+      },
+      $source = `'vitest'`,
+      $vi = `vi`,
+      if ($sym <: not `jest`) {
+        $sym <: ensure_import_from($source)
+      } else {
+        $vi <: ensure_import_from($source)
+      }
+    }
+  }
+}
+
+pattern main_migration() {
+  file($body) where {
+    $body <: contains bubble or {
+      `JEST_WORKER_ID` => `VITEST_POOL_ID`,
+
+      // Imports
+      `import $_ from 'jest'` => .,
+
+      // MODLE MOCKS
+      // default literal mocking
+      `jest.mock($module, $mockImplementation)` where {
+        $mockImplementation <: arrow_function($body) where {
+            // still waiting for default mock confirmation
+            $body <: literal_value() => `({ 
+              default: $body 
+            })`
+          }
+        } => `vi.mock($module, $mockImplementation)`,
```

Comment:
This needs to support traditional functions.

Reply:
Thanks @morgante we will update pattern to support traditional functions.